tests/testthat/test-listing_dsl01.R

```diff
@@ -0,0 +1,88 @@
+library(dplyr)
+
+result <- formatters::ex_adsl %>%
+  mutate(
+    ID = paste(SITEID, SUBJID, sep = "/"),
+    ASR = paste(AGE, SEX, RACE, sep = "/"),
+    SSADM = toupper(format(as.Date(TRTSDTM), format = "%d%b%Y")),
+    STDWD = as.numeric(ceiling(difftime(TRTEDTM, TRTSDTM, units = "days"))),
+    DISCONT = ifelse(!is.na(DCSREAS) & toupper(EOSSTT) == "DISCONTINUED", "Yes", "No")
+  ) %>%
+  select(ID, ASR, ARMCD, SSADM, STDWD, DISCONT)
+
+formatters::var_labels(result) <- c(
+  ID = "Center/Patient ID",
+  ASR = "Age/Sex/Race",
+  ARMCD = "Treatment",
+  SSADM = "Date of First\nStudy Drug\nAdministration",
+  STDWD = "Study Day\nof Withdrawal",
+  DISCONT = "Discontinued\nEarly from Study?"
+)
+
+testthat::test_that("DSL01 listing is produced correctly", {
+  lsting <- testthat::expect_message(as_listing(
+    result,
+    cols = names(result),
+    main_title = "Listing of Patients with Study Drug Withdrawn Due to Adverse Events"
+  ), "sorting incoming data by key columns")
+
+  result_matrix <- matrix_form(head(lsting, 10))
+  expected_strings <- structure(
```

Comment:
please change this to a call to `matrix` rather than the roundabout `structure` based construction